Changelog updates for 7.2 (#3291)
ROCm/composable_kernel: CHANGELOG.md

* Added an optional template parameter `Arch` (`gfx9_t`, `gfx12_t` etc.) to `make_kernel` to support linking multiple object files that have the same kernel compiled for different architectures.
* FMHA examples and tests can be built for multiple architectures (gfx9, gfx950, gfx12) at the same time.

### Upcoming changes

* Composable Kernel will be adopting C++20 features in an upcoming ROCm release, updating the minimum compiler requirement to C++20. Ensure that your development environment complies with this requirement to facilitate a seamless transition.

> * Starting with the next major ROCm release, Composable Kernel will transition to a header-only library.

Reviewer: ```suggestion
* In an upcoming major ROCm release, Composable Kernel will transition to a header-only library.
```

>   By default, static libraries and ckProfiler will no longer be built.

Reviewer: ```suggestion
Neither ckProfiler nor the static libraries will be packaged with Composable Kernel nor built by default with Composable Kernel. ckProfiler can be built independently from Composable Kernel as a standalone binary, and the static Composable Kernel libraries can be built from source.
```

  However, users who require static libraries for integrating Composable Kernel into their products will still have the option to build them.

>   Additionally, ckProfiler can be built as a standalone binary or packaged for installation as needed.

Reviewer: I rewrote these two lines above.
